src/test/java/org/cactoos/http/io/BlockingSocketServerTest.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.http.io;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.SocketTimeoutException;
+import org.cactoos.text.TextOf;
+import org.hamcrest.core.IsEqual;
+import org.junit.Test;
+import org.llorllale.cactoos.matchers.Assertion;
+import org.llorllale.cactoos.matchers.MatcherOf;
+import org.llorllale.cactoos.matchers.Throws;
+
+/**
+ * Test case for {@link BlockingSocketServer}.
+ *
+ * @since 0.1
+ * @checkstyle JavadocMethodCheck (500 lines)
+ * @checkstyle JavadocVariableCheck (500 lines)
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+public final class BlockingSocketServerTest {
+
+    // @checkstyle MagicNumberCheck (1 line)
+    @Test(timeout = 1000)
```

Comment:
@fabriciofx I don't think it is a problem, when the test is passing, it should finish in 100ms more or less (see the timeout of the connect). If it doesn't, it means there is a bug and the test shouldn't pass.
We do need the timeout on test to prevent completely blocking everything in case there is a big bug and the timeout from the connect doesn't happen.

An alternative would be to rely on Clock and co to test it with a fake clock, but the problem is that we are testing things that do not take Clock into account, so it would mean replacing the whole socket lib of Java with something else. It would be a bit too much knowing that this timeout will never be a problem since the test has 10x more time (1000ms) to wait than the expected running time (~100ms).

I can reduce the connect to 10ms if you prefer though.

Reply:
@victornoel right, let's reduce the timeout to 10ms, as suggested by you.